internal/settings/cache.go

```diff
@@ -23,10 +24,16 @@ func cacheKey(key string) string {
 }
 
 func setCache[T any](key string, ttl int64, value T) error {
-	entry := Entry[T]{Data: value}
+	exp := int64(0)
 	if ttl > 0 {
-		entry.Expiration = time.Now().Unix() + ttl
+		exp = time.Now().Unix() + ttl
 	}
```

Comment:
Why not just `exp := time.Now().Unix() + ttl`. That would be preserving the previous behaviour. Setting exp to 0 for ttl = 0 changes how this case is handled in getCache.

Reply:
I wanted non-expiring entries as well.  I don't remember why I wanted them.
And we don't have any, so I'll remove this.

Reply:
Currently entry.Expiration == 0 means returning nil in getCache.